crates/nu-command/src/network/http/patch.rs

```diff
@@ -0,0 +1,199 @@
+use nu_protocol::{
+    Category, Example, PipelineData, ShellError, Signature, SyntaxShape, Type, Value,
+};
+
+use crate::network::http::client::{
+    http_client, http_parse_url, request_add_authorization_header, request_add_custom_headers,
+    request_handle_response, request_set_body, request_set_timeout,
+};
+
+#[derive(Clone)]
+pub struct SubCommand;
+
+impl Command for SubCommand {
+    fn name(&self) -> &str {
+        "http patch"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build("http patch")
+            .input_output_types(vec![(Type::Nothing, Type::Any)])
+            .allow_variants_without_examples(true)
+            .required("URL", SyntaxShape::String, "the URL to post to")
+            .required("data", SyntaxShape::Any, "the contents of the post body")
+            .named(
+                "user",
+                SyntaxShape::Any,
+                "the username when authenticating",
+                Some('u'),
+            )
+            .named(
+                "password",
+                SyntaxShape::Any,
+                "the password when authenticating",
+                Some('p'),
+            )
+            .named(
+                "content-type",
+                SyntaxShape::Any,
+                "the MIME type of content to post",
+                Some('t'),
+            )
+            .named(
+                "content-length",
+                SyntaxShape::Any,
+                "the length of the content being posted",
+                Some('l'),
+            )
+            .named(
+                "max-time",
+                SyntaxShape::Int,
+                "timeout period in seconds",
+                Some('m'),
+            )
+            .named(
+                "headers",
+                SyntaxShape::Any,
+                "custom headers you want to add ",
+                Some('H'),
+            )
+            .switch(
+                "raw",
+                "return values as a string instead of a table",
+                Some('r'),
+            )
+            .switch(
+                "insecure",
+                "allow insecure server connections when using SSL",
+                Some('k'),
+            )
+            .filter()
+            .category(Category::Network)
+    }
+
+    fn usage(&self) -> &str {
+        "Patch a body to a URL."
+    }
+
+    fn extra_usage(&self) -> &str {
+        "Performs HTTP PATCH operation."
+    }
+
+    fn search_terms(&self) -> Vec<&str> {
+        vec!["network", "send", "push"]
+    }
+
+    fn run(
+        &self,
+        engine_state: &EngineState,
+        stack: &mut Stack,
+        call: &Call,
+        input: PipelineData,
+    ) -> Result<PipelineData, ShellError> {
+        run_patch(engine_state, stack, call, input)
+    }
+
+    fn examples(&self) -> Vec<Example> {
+        vec![
+            Example {
+                description: "Patch content to url.com",
+                example: "http patch url.com 'body'",
```

Comment:
Having only url.com gives the impression that you don't need a fully qualified url. I tried it and it pukes like this.

Reply:
I updated the examples.